test/test_device_batch_memcpy.cu

```diff
@@ -0,0 +1,548 @@
+    << ", src size: " << stats_src_offsets << ", dst size: " << stats_dst_offsets               //
+    << ", sizes size: " << stats_sizes << ", cpy_data_size: " << stats_data_copied              //
+    << ", total: " << (stats_src_offsets + stats_dst_offsets + stats_sizes + stats_data_copied) //
+    << ", duration: " << duration                                                               //
+    << ", BW: "
+    << ((double)(stats_src_offsets + stats_dst_offsets + stats_sizes + stats_data_copied) /
+        1000000000.0) /
+         (duration / 1000.0)
+    << "GB/s \n";
+#endif
+
+  for (ByteOffsetT i = 0; i < num_total_bytes; i++)
+  {
+    if (h_gpu_results[i] != h_out[i])
+    {
+      std::cout << "Mismatch at index " << i
+                << ", CPU vs. GPU: " << static_cast<uint16_t>(h_gpu_results[i]) << ", "
+                << static_cast<uint16_t>(h_out[i]) << "\n";
+    }
+    AssertEquals(h_out[i], h_gpu_results[i]);
+  }
+
+  CubDebugExit(cudaFree(d_in));
+  CubDebugExit(cudaFree(d_out));
+  CubDebugExit(cudaFree(d_buffer_src_offsets));
+  CubDebugExit(cudaFree(d_buffer_dst_offsets));
+  CubDebugExit(cudaFree(d_buffer_sizes));
+  CubDebugExit(cudaFree(d_temp_storage));
+}
+
+template <int LOGICAL_WARP_SIZE, typename VectorT, typename ByteOffsetT>
+__global__ void TestVectorizedCopyKernel(const void *d_in, void *d_out, ByteOffsetT copy_size)
+{
+  cub::detail::VectorizedCopy<LOGICAL_WARP_SIZE, VectorT>(threadIdx.x, d_out, copy_size, d_in);
+}
+
+struct TupleMemberEqualityOp
+{
+  template <typename T>
+  __host__ __device__ __forceinline__ bool operator()(T tuple)
+  {
+    return thrust::get<0>(tuple) == thrust::get<1>(tuple);
+  }
+};
+
+/**
+ * @brief Tests the VectorizedCopy for various aligned and misaligned input and output pointers.
+ * @tparam VectorT The vector type used for vectorized stores (i.e., one of uint4, uint2, uint32_t)
+ */
+template <typename VectorT>
+void TestVectorizedCopy()
+{
+  // Prepare data
+  constexpr std::size_t alloc_size = 1024 * 1024;
+  thrust::device_vector<char> data_in(alloc_size);
+  thrust::device_vector<char> data_out(alloc_size);
+  thrust::sequence(data_in.begin(), data_in.end(), static_cast<char>(0));
+  thrust::fill_n(data_out.begin(), alloc_size, static_cast<char>(0x42));
+
+  constexpr uint32_t threads_per_block = 8;
+
+  int test_count = 0;
+  auto d_in      = thrust::raw_pointer_cast(data_in.data());
+  auto d_out     = thrust::raw_pointer_cast(data_out.data());
+
+  std::vector<std::size_t> in_offsets{0, 1, sizeof(uint32_t) - 1};
+  std::vector<std::size_t> out_offsets{0, 1, sizeof(VectorT) - 1};
+  std::vector<std::size_t> copy_sizes{0,
+                                      1,
+                                      sizeof(uint32_t),
+                                      sizeof(VectorT),
+                                      2 * threads_per_block * sizeof(VectorT)};
+  for (auto copy_sizes_it = std::cbegin(copy_sizes); copy_sizes_it < std::cend(copy_sizes);
```

Comment:
`cbegin` and `cend` are C++14 features. We have to build with C++11.

Reply:
Thanks. Fixed 👍